### PR TITLE
fix input generator for falcon40b

### PR DESCRIPTION
**Pull request by eaidova**
# What does this PR do?
this is temporal workaround until we can not use https://github.com/huggingface/optimum/pull/1825

falcon dummy input generator is not able to handle falcon-40b model (case with new_decoder_architecture and multi_query)

### CODE REVIEW

**Comment by eaidova** [comment]
@echarlaix could you please take a look?

**Review by AlexKoff88 — APPROVED** [review]

File: optimum/exporters/openvino/model_configs.py [hunk 1]
```diff
@@ -437,3 +438,52 @@ class OrionOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
...
+            batch_size=batch_size,
+            sequence_length=sequence_length,
+            random_batch_size_range=random_batch_size_range,
+            random_sequence_length_range=random_sequence_length_range,
+            **kwargs,
+        )
+        if normalized_config.new_decoder_architecture and normalized_config.multi_query:
+            self.num_kv_heads = normalized_config.num_attention_heads
+        elif normalized_config.new_decoder_architecture and not normalized_config.multi_query:
+            self.num_kv_heads = normalized_config.num_kv_heads
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_configs.py:466]: can we simplify like 
```suggestion
        if normalized_config.new_decoder_architecture:
            self.num_kv_heads = normalized_config.num_attention_heads if normalized_config.multi_query else normalized_config.num_kv_heads
```

**Review by echarlaix — APPROVED** [review]

File: optimum/exporters/openvino/model_configs.py [hunk 2]
```diff
@@ -437,3 +438,52 @@ class OrionOpenVINOConfig(TextDecoderWithPositionIdsOnnxConfig):
...
+        else:
+            self.num_kv_heads = 1
```
Comment (echarlaix) [inline optimum/exporters/openvino/model_configs.py:468]: also shouldn't it be 
```suggestion
        if normalized_config.new_decoder_architecture:
            self.num_kv_heads = normalized_config.num_attention_heads
        else:
            self.num_kv_heads = normalized_config.num_kv_heads if not normalized_config.multi_query else 1
```
?

Comment (eaidova) [reply optimum/exporters/openvino/model_configs.py]: this variant is correct

**Pull request merged**